unittest/server/src/ivi_controller_uinttests.cpp

```diff
@@ -0,0 +1,3361 @@
+    SET_RETURN_SEQ(wl_resource_create, &screenshot[0], 1);
+
+    controller_surface_screenshot(nullptr, nullptr, 1, 1);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_destroy_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_surface_screenshot_errByResult
+ * @brief               Test case of controller_surface_screenshot() where surface_get_size() fails, return null pointer
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Mocking the wl_resource_create() does return an object
+ *                      -# Mocking the get_surface_from_id() does return an object
+ *                      -# Set the surface_get_size() does return null pointer
+ *                      -# Calling the controller_surface_screenshot()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ *                         +# wl_resource_destroy() must be called once time
+ */
+TEST_F(ControllerTests, controller_surface_screenshot_errByResult)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)&mp_iviController[0], 1);
+    struct wl_resource * screenshot[1];
+    SET_RETURN_SEQ(wl_resource_create, &screenshot[0], 1);
+    struct ivi_layout_surface *l_layout_surface[1];
+    SET_RETURN_SEQ(get_surface_from_id, &l_layout_surface[0], 1);
+
+    surface_get_size_fake.custom_fake = nullptr;
+
+    controller_surface_screenshot(nullptr, nullptr, 1, 1);
+
+    ASSERT_EQ(wl_resource_post_event_fake.call_count, 1);
+    ASSERT_EQ(wl_resource_destroy_fake.call_count, 1);
+}
+
+/** ================================================================================================
+ * @test_id             controller_surface_screenshot_errReadpixByResult
+ * @brief               Test case of controller_surface_screenshot() where surface_dump() fails, return -1
+ * @test_procedure Steps:
+ *                      -# Mocking the wl_resource_get_user_data() does return an object
+ *                      -# Mocking the wl_resource_create() does return an object
+ *                      -# Mocking the get_surface_from_id() does return an object
+ *                      -# Set the surface_get_size() does return null pointer
+ *                      -# Mocking the surface_dump() does return -1
+ *                      -# Calling the controller_surface_screenshot()
+ *                      -# Verification point:
+ *                         +# wl_resource_post_event() must be called once time
+ *                         +# wl_resource_destroy() must be called once time
+ */
+TEST_F(ControllerTests, controller_surface_screenshot_errReadpixByResult)
+{
+    SET_RETURN_SEQ(wl_resource_get_user_data, (void**)&mp_iviController[0], 1);
+    struct wl_resource * screenshot[1];
```

Comment:
As a discussion: the variable only declare without initial, it may make a potential issue when getting random value. Please take a look to all variables without initialize.